posthog/src/test/java/com/posthog/internal/PostHogApiTest.kt

```diff
@@ -0,0 +1,111 @@
+package com.posthog.internal
+
+import com.posthog.PostHogConfig
+import com.posthog.apiKey
+import com.posthog.generateEvent
+import com.posthog.responseApi
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.Assert.assertThrows
+import org.junit.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+
+internal class PostHogApiTest {
+
+    private fun getSut(
+        host: String = "https://app.posthog.com",
+    ): PostHogApi {
+        val config = PostHogConfig(apiKey, host)
+        val serializer = PostHogSerializer(config)
+        return PostHogApi(config, serializer)
+    }
+
+    private fun mockHttp(
+        response: MockResponse = MockResponse()
+            .setBody(""),
+    ): MockWebServer {
+        val mock = MockWebServer()
+        mock.start()
+        mock.enqueue(response)
+        return mock
+    }
+
+    @Test
+    fun `batch returns successful response`() {
+        val http = mockHttp()
+        val url = http.url("/")
+
+        val sut = getSut(host = url.toString())
```

Comment:
As a style thing I'd have

`val api = getPostHogAPI(bla)` here
or
`val sut = getPostHogAPI(bla)`

I _know_ it's the system under test from the context and find myself having to consciously translate it

It's only nit picking though